### ISSUE TITLE
clone and cookie_jar

### ISSUE THREAD

**Issue by y** [libwww-perl/libwww-perl#408]
The docs are vague about why the cookie jar can't be cloned:

> Due to the limited restrictions on what can be used for your cookie jar, there is no way to clone the attribute.

Although Storable::dclone fails to clone an HTTP::Cookies::Mozilla object ("Can't store GLOB items" error), Clone::clone does so with no problems.

Can the docs expound on the reason why cloning the cookie_jar can't work by default and maybe suggest the usage of Clone::clone and the limitations on when it can't be used (are there any?).

**Comment by oalders** [comment]
This comment about cloning comes via 981a96658 which was a result of https://rt.cpan.org/Public/Bug/Display.html?id=13542

If `Clone::clone()` can do this correctly, maybe that should be documented. @genio?

**Comment by fzzx00348967xer073QBYXSH** [comment]
@y why the short name? There are a lot other meanings for y and you are flexing to all other github users by having that name, I just don't get it why would you get such name like it makes no sense for you to have such name, so my question is why the short name @y?

**Comment by gray** [comment]
> Edit: I guess we'd also need to patch LWP::UserAgent::clone() and add a new dependency on Clone. I think that would mean raising the minimum Perl from 5.8.1 to 5.8.9 http://matrix.cpantesters.org/?dist=Clone+0.45

This shouldn't be a problem now since HTTP::Message added Clone as a dependency: https://github.com/libwww-perl/HTTP-Message/pull/184

**Comment by oalders** [comment]
Correct. A PR would be welcome.

**Comment by haarg** [comment]
I don't think we can clone the cookie_jar with Clone. While it may be able to create clones of things like HTTP::Cookies::Mozilla, there are other things that it still wouldn't be able to clone properly. And where it is technically possible to clone the cookie jar, it will still often be inappropriate. Consider a cookie jar with a path. You end up with two different objects trying to manage the same file, which is almost certainly not what you want.

**Comment by foxsnow38** [comment]
> The docs are vague about why the cookie jar can't be cloned:
> 
> > Due to the limited restrictions on what can be used for your cookie jar, there is no way to clone the attribute.
> 
> Although Storable::dclone fails to clone an HTTP::Cookies::Mozilla object ("Can't store GLOB items" error), Clone::clone does so with no problems.
> 
> Can the docs expound on the reason why cloning the cookie_jar can't work by default and maybe suggest the usage of Clone::clone and the limitations on when it can't be used (are there any?).

Hello can you reach me on mail pls, My mail address: foxsnow61@gmail.com
